src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -414,16 +404,22 @@ private static SslException CreateSslException(IntPtr sslPtr, string message, in
 
         private sealed class SslException : Exception
         {
-            public SslException(string message)
-                : base(message + ": " + Marshal.PtrToStringAnsi(libssl.ERR_reason_error_string(libssl.ERR_get_error())))
+            private string message;
+            public override string Message { get { return message; } }
+            
+            public SslException(string inputMessage)
+                : base()
             {
-                HResult = (int)libssl.ERR_get_error();
+                ulong errorVal = libssl.ERR_get_error();
+                HResult = (int)errorVal;
+                this.message = inputMessage + ": " + Marshal.PtrToStringAnsi(libssl.ERR_reason_error_string(errorVal));
             }
 
-            public SslException(string message, int error)
-                : base(message + ": " + error)
+            public SslException(string inputMessage, int error)
+                : base()
             {
                 HResult = error;
+                this.message = inputMessage + ": " + error;
```

Comment:
These string concats will need to be moved into the .resx file as strings with placeholders like {0} in them.